Assets/Scripts/Player/PlayerHandDetection.cs

```diff
@@ -3,19 +3,74 @@
 using UnityEngine;
 
 public class PlayerHandDetection : MonoBehaviour {
+	[SerializeField] private GameObject interactArrowPrefab;
+	[SerializeField] private PlayerCarrying playerCarrying;
 	private Interactable detectedInteractItem;
+	private List<Interactable> interactableList = new List<Interactable>();
+	private GameObject interactArrow;
+	private InteractArrow interactArrowScript;
+
 	public Interactable DetectedInteractItem => detectedInteractItem;
 
-	public void OnTriggerStay2D(Collider2D collider) {
-		Interactable interactable = collider.GetComponent<Interactable>();
+	private void Start() {
+		interactArrow = Instantiate(interactArrowPrefab);
+		interactArrowScript = interactArrow.GetComponent<InteractArrow>();
+		interactArrow.SetActive(false);
+	}
+
+	private void OnTriggerEnter2D(Collider2D other) {
+		Interactable interactable = other.GetComponent<Interactable>();
 		if (interactable != null) {
-			detectedInteractItem = interactable;
+			interactableList.Add(interactable);
 		}
 	}
 
-	public void OnTriggerExit2D(Collider2D collider) {
-		if (collider.GetComponent<Interactable>() != null) {
+	private void Update() {
+		if (interactableList.Count == 0)
+			return;
+
+		SetClosestItem();
+	}
+
+	private void OnTriggerExit2D(Collider2D other) {
+		Interactable interactable = other.GetComponent<Interactable>();
+		if (interactable != null) {
+			interactableList.Remove(interactable);	
+			
+			if (interactableList.Count == 0) {
+				detectedInteractItem = null;
+				interactArrow.SetActive(false);
+			}
+			else {
+				detectedInteractItem = null;
+				SetClosestItem();
+			}
```

Comment:
Can't this be changed?
```suggestion
detectedInteractItem = null;
			if (interactableList.Count == 0)
				interactArrow.SetActive(false);
```